R/interpret.R

```diff
@@ -0,0 +1,147 @@
+  mi <- extract_model_info(x)
+  if (mi$best_model_name != "glmnet")
+    warning("Interpreting glmnet model, but ", mi$best_model_name, " performed ",
+            "best in cross-validation and will be used to make predictions. ",
+            "To use the glmnet model for predictions, extract it with ",
+            "x['glmnet'].")
+
+  g <- x$glmnet$finalModel
+  coefs <- stats::coef(g)
+
+  # If user didn't specify sparsity, use best lambda
+  if (is.null(sparsity)) {
+    lambda <- g$lambdaOpt
+    coefs <- stats::coef(g, s = lambda)
+  } else {
+    if (!is.numeric(sparsity) || sparsity < 0 || sparsity > 1)
+      stop("sparsity must be a numeric value between 1 and 100")
+    j <- round(stats::quantile(seq_len(ncol(coefs)), sparsity), 0)
+    coefs <- coefs[, j, drop = FALSE]
+    lambda <- g$lambda[j]
+  }
+  coefs <-
+    tibble::tibble(variable = rownames(coefs), coefficient = coefs[, 1]) %>%
+    dplyr::arrange(desc(variable == "(Intercept)"), desc(abs(coefficient)))
+
+  if (remove_zeros)
+    coefs <- dplyr::filter(coefs, coefficient != 0)
+
+  structure(coefs,
+            class = c("interpret", class(coefs)),
+            m_class = mi$m_class,
+            target = mi$target,
+            lambda = lambda,
+            alpha = mi$best_model_tune$alpha)
+}
+
+#' Plot regularized model coefficients
+#'
+#' @param x A \code{interpret} object or a data frame with columns "variable"
+#'   and "coefficient"
+#' @param include_intercept If FALSE (default) the intercept estimate will not
+#'   be plotted
+#' @param remove_zeros If FALSE (default) estimates of 0 are included in the
+#'   plot. Note that \code{\link{interpret}} removes zero estimates by default
+#' @param title Plot title. NULL for no title; character for custom title. If
+#'   left blank contains the model class and outcome variable
+#' @param caption Plot caption, appears in lower-right. NULL for no caption;
+#'   character for custom caption. If left blank the caption will contain info
+#'   including the hyperparameter values of the model used by
+#'   \code{\link{interpret}} to determine coefficient estimates.
+#' @param font_size Relative size of all fonts in plot, default = 11
+#' @param point_size Size of dots, default = 3
+#' @param print Print the plot? Default = TRUE
+#' @param ... Unused
+#'
+#'
+#' @return A ggplot object, invisibly.
+#' @export
+#' @seealso \code{\link{interpret}}
+#'
+#' @examples
+#' machine_learn(mtcars, outcome = mpg, models = "glm", tune = FALSE) %>%
+#'   interpret() %>%
+#'   plot(font_size = 14)
+plot.interpret <- function(x, include_intercept = FALSE, remove_zeros = FALSE,
+                       title, caption, font_size = 11, point_size = 3,
+                       print = TRUE, ... ) {
+
+  ats <- attributes(x)
+
+  if ( (is.data.frame(x) && names(x) != c("variable", "coefficient") ) ||
+       !is.data.frame(x))
+    stop("x must be a data frame from interpret, or at least look like one!")
+
+  if (remove_zeros)
```

Comment:
The remove zeros argument here might be confusing if someone wants them in the plot but doesn't know that interpret will get rid of them by default. Not a big deal.